packages/http-client/src/document.ts

```diff
@@ -1,63 +1,52 @@
-import {
-  CeramicCommit, Context, DocOpts, DocParams, DocState, Doctype, DoctypeConstructor, DoctypeUtils
-} from "@ceramicnetwork/common"
-
-import DocID, { CommitID } from '@ceramicnetwork/docid';
-
-import { fetchJson, typeDocID, delay } from './utils'
-import { CeramicClientConfig } from "./ceramic-http-client"
-
-export class Document extends Doctype {
-
-  private _syncEnabled: boolean
-  private readonly _syncInterval: number
-
-  public doctypeLogic: DoctypeConstructor<Doctype>
+import { BehaviorSubject, Observable, timer } from 'rxjs'
+import { throttle } from 'rxjs/operators'
+import { CeramicCommit, DocOpts, DocState, DoctypeUtils, RunningStateLike } from '@ceramicnetwork/common';
+import { DocID, CommitID } from '@ceramicnetwork/docid';
+import { fetchJson } from './utils'
+
+export class Document extends Observable<DocState> implements RunningStateLike {
+  private readonly state$: BehaviorSubject<DocState>;
+
+  constructor (initial: DocState, private _apiUrl: string, docSyncInterval: number) {
+    super(subscriber => {
+      const periodicUpdates = timer(0, docSyncInterval).pipe(throttle(() => this._syncState())).subscribe();
```

Comment:
Is it still possible to disable syncing here like before?

Reply:
Maybe `0` means disabled?

Reply:
Syncing now has explicit opt-in semantics (API semantics by @stbrody). There is no global `sync all` or `does not sync anything` flag. It does not make sense anymore. Adding "no-sync" flag to opt-in semantics seems confusing.

Reply:
Aha, yes ofc! Thanks for clarifying